scripts/upload.py

```diff
@@ -0,0 +1,75 @@
+
+import os
+
+from boto.s3.connection import S3Connection
+from boto.s3.key import Key
+import click
+
+import creds
+
+
+def get_or_create_bucket(conn, bucket_name):
+    """Get or create an S3 bucket.
+
+    :param conn: boto.s3.connection.S3Connection
+    :param bucket_name: string
+    :returns: boto.s3.bucket.Bucket
+    """
+    try:
+        bucket = conn.get_bucket(bucket_name)
+    except ValueError:
+        bucket = conn.create_bucket(bucket_name)
+
+    return bucket
+
+
+def sizeof_fmt(num):
+    """Human readable file size.
+
+    Modified from http://stackoverflow.com/a/1094933/1377021
+
+    :param num: float
+    :returns: string
+    """
+    for unit in ['', 'k', 'm', 'g', 't', 'p', 'e', 'z']:
+        if abs(num) < 1024.0:
+            return "%.0f%s%s" % (num, unit, 'b')
+        num /= 1024.0
+
+    return "%.f%s%s" % (num, 'y', 'b')
+
+
+@click.command()
+@click.argument('directory', type=click.Path(exists=True))
+def upload(directory):
+    """Upload a directory to S3.
+
+    DIRECTORY: Directory to upload. Required.
+    """
+    conn = S3Connection(creds.AWS_ACCESS_KEY_ID, creds.AWS_SECRET_ACCESS_KEY)
```

Comment:
I think it would be better to not specify the credentials, and just stick to using the default boto credential loading mechanisms.